### ISSUE TITLE
backends/gr.jl needs :match_dimensions added

### ISSUE THREAD

**Issue by JeffFessler** [JuliaPlots/Plots.jl#1694]
As discussed in #273 and #196, the option `transpose=true` should help those of us who like f(x,y) not f(y,x).  (I work with a lot of 3D images and strongly prefer f(x,y,z) to f(y,x,z)...)
However, the `transpose=true` option currently throws a warning with the GR backend:
`Warning: Keyword argument match_dimensions not supported with Plots.GRBackend`

It can be fixed simply by adding the keyword `:match_dimensions` to backends/gr.jl in the long list of supported keywords there.

Here's a MWE that generates the warning:  
`using Plots # uses gr() by default`
`x=-10:10; y=-3:3; f=x.^2 .+ 0*y'`
`heatmap(x, y, f, transpose=true)`

I'm using Plots v0.19.3 with Julia 0.7.0

**Comment by mkborregaard** [comment]
Absolutely. In the mood for a quick PR? :-p

**Comment by JeffFessler** [comment]
Sure.  It wasn't so quick because it's my first PR (!) so hopefully I did it correctly...
The CI tests failed it seems but the failures didn't seem related to my tiny modification...

**Comment by mkborregaard** [comment]
Yay! welcome to the wonderful world of collaborative open source development!
Do you know how to rebase your PR on current master?

**Comment by JeffFessler** [comment]
I did my one-line edit and PR in the browser.  The rebase instructions seem to be for someone who has a local copy of the repo on their host:
https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request
Is there a way to rebase a PR on the website in the browser?
Maybe it would be easier to make my one-line change as a new PR on the current master?

**Comment by daschw** [comment]
Are you talking about #1696 ? Why would this need to be rebased, @mkborregaard ?

**Comment by mkborregaard** [comment]
Yes, exactly, it was just to run the tests. But I think it's fine. Thank you @JeffFessler !

**Comment by daschw** [comment]
@JeffFessler, honestly I don't know how to do this in the browser. To do this locally, you can open julia and run
```julia
(v1.0) pkg> dev Plots
```
this clones the current master of Plots in your julia dev directory. Head over to this directory and add your fork as a remote:
```julia
julia> cd(joinpath(homedir(), ".julia", "dev", "Plots"))
shell> git remote add forked git@github.com:JeffFessler/Plots.jl.git
```
Fetch origin and your fork and checkout the branch you want to rebase:
```julia
shell> git fetch origin
shell> git fetch forked
shell> git checkout patch-1
```
Now rebase and push with
```
shell> git rebase origin/master
shell> git push -f forked patch-1
```


... apparently I was too slow, but maybe this helps for the future.

**Comment by mkborregaard** [comment]
I don't know any of that, I just click and drag in GitKraken :-)

<img width="1015" alt="skaermbillede 2018-08-26 kl 23 04 43" src="https://user-images.githubusercontent.com/8429802/44632976-95467500-a984-11e8-8669-e6b88821b083.png">

(no you weren't too slow, @daschw I just thought it was a bit much of me to ask for a small change)

**Comment by mkborregaard** [comment]
Here's another neat trick for next time ;-) : if you end the title of your PR with the words "fixes #1694" , the issue will automatically close when the PR is merged.

**Comment by mkborregaard** [comment]
That feels rather like magic

**Comment by wsshin** [comment]
Is this feature still available?  I tried the example shown in https://github.com/JuliaPlots/Plots.jl/issues/1694#issue-354018322,  but neither `transpose=true` nor `match_dimensions=true` works.

**Comment by JeffFessler** [comment]
Oh dear, I just tried the example on Julia 1.6 and 1.5 using Plots 1.11.1 (on a Mac) and in both cases the code ran and produced an output image but that image was incorrect.  It doesn't even have the correct number of pixels in it.  So something has changed.  I hadn't noticed because I wrote my own wrapper around heatmap that has a transpose built into it, like `heatmap(x, y, f')` and that works fine.
https://github.com/JeffFessler/MIRTjim.jl

**Comment by wsshin** [comment]
`transverse` and `match_dimensions` are not even valid attributes:
```
(@v1.6) pkg> st Plots
      Status `~/.julia/environments/v1.6/Project.toml`
  [91a5bcdd] Plots v1.11.1

julia> plotattr("transverse")
ERROR: There is no attribute named transverse

julia> plotattr("match_dimensions")
ERROR: There is no attribute named match_dimensions
```
How can we track down when they disappeared?

**Comment by JeffFessler** [comment]
In v1.11.1, the only place where `match_dimensions` appears is in
https://github.com/JuliaPlots/Plots.jl/blob/c0824bdc628345b4859ebcafc437cbfd63e63ccb/src/arg_desc.jl

In earlier versions, like v1.5.8, it also appeared in several other files: utils.jl backends.jl args.jl
I've searched the Issues and the PRs for `match_dimensions` and could not find any discussion of it, but I don't know how thorough the search is in github...

**Comment by goretkin** [comment]
This looks like the relevant change: https://github.com/JuliaPlots/Plots.jl/pull/3056

**Comment by goretkin** [comment]
I am not sure how to get the previous functionality. @daschw , is there a replacement?

By the way, `Surface` is defined here:
https://github.com/JuliaPlots/RecipesPipeline.jl/blob/00ed890cb63f5d65183f0a10f4172f5124a2f746/src/utils.jl#L72-L75
